components/apimgt/org.wso2.carbon.apimgt.rest.api.commons/src/main/java/org/wso2/carbon/apimgt/rest/api/common/impl/BasicAuthAuthenticator.java

```diff
@@ -77,10 +77,7 @@ public boolean authenticate(Request request, Response responder, ServiceMethodIn
 
     private boolean authenticate(String username, String password) {
         //todo improve
-        if (username.equals(password)) {
-            return true;
-        }
-        return false;
+        return username.equals(password);
```

Comment:
Here we need to implement the authentication logic when the user name and password are provided.
Need to check for an API from IS.
Issue created. https://github.com/wso2/carbon-apimgt/issues/4323